### PR TITLE
adding docstrings to cfn.py

### PR DESCRIPTION
**Pull request by joshuaroot**
* adding docstrings to cfn.py

### CODE REVIEW

File: c7n/resources/cfn.py [hunk 1]
```diff
@@ -37,6 +37,25 @@ class CloudFormation(QueryResourceManager):
...
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: cloudformation-delete-failed-stacks
+                resource: cfn
+                filters:
+                  - type: value
```
Comment (kapilt) [inline c7n/resources/cfn.py]: StackStatus: CREATE_FAILED

is equivalent filter

**Comment by kapilt** [comment]
the test here is extraneous, please remove.

**Comment by joshuaroot** [comment]
docstrings & examples here have been checked for validity/confirmed functional

**Comment by kapilt** [comment]
thanks, this still has the data files from the removed test that can also be removed.

**Pull request merged**